src/foundry/client-esm/applications/apps/permission-config.d.mts

```diff
@@ -1,4 +1,42 @@
+// import type { UserPermission } from "../../../common/constants.d.mts";
+import type { UserPermission } from "../../../common/constants.d.mts";
+import type { CONST } from "../../client.d.mts";
+import type { ApplicationConfiguration, FormFooterButton } from "../_types.d.mts";
 import type ApplicationV2 from "../api/application.d.mts";
 import type HandlebarsApplicationMixin from "../api/handlebars-application.mjs";
 
-export default class PermissionConfig extends HandlebarsApplicationMixin(ApplicationV2) {}
+// TODO: decide if we want to keep this as part of _prepareContext return
+type PermissionWithRoles = UserPermission & {
+  id: string;
+  roles: {
+    name: string;
+    value: boolean;
+    readonly: string;
+  }[];
+};
+
+type PermissionConfigRenderContext = {
```

Comment:
I would make this an interface and stick it inside the PermissionConfig namespace, probably simplify to "RenderContext" as the name; the idea is someone might want to extend this application class and be able to nicely access the types in the `super` call.

Reply:
Also make it an interface while you're at it so users can extend it.